src/tests/Microsoft.Diagnostics.NETCore.Client/GetProcessInfoTests.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    public class GetProcessInfoTests
+    {
+        private readonly ITestOutputHelper output;
+
+        public GetProcessInfoTests(ITestOutputHelper outputHelper)
+        {
+            output = outputHelper;
+        }
+
+        [Fact]
+        public void BasicProcessInfoTest()
+        {
+            using TestRunner runner = new TestRunner(CommonHelper.GetTraceePathWithArgs(targetFramework: "net5.0"), output);
+            runner.Start();
+
+            try
+            {
+                DiagnosticsClient client = new DiagnosticsClient(runner.Pid);
+
+                ProcessInfo processInfo = client.GetProcessInfo();
```

Comment:
@josalem, seems invoking the GetProcessInfo2 command crashes the target runtime very frequently. Here's the callstack from the perspective of the test (it's in the midst of reading the IpcHeader when the runtime instance crashes):

```
    BinaryReader.InternalRead(Int32 numBytes)
    BinaryReader.ReadUInt16()
    IpcHeader.TryParse(BinaryReader reader) line 58
    IpcMessage.Parse(Stream stream) line 114
    IpcClient.Read(Stream stream) line 60
    IpcClient.SendMessage(IpcEndpoint endpoint, IpcMessage message) line 28
    DiagnosticsClient.TryGetProcessInfo2(ProcessInfo& processInfo) line 220
    DiagnosticsClient.GetProcessInfo() line 198
    GetProcessInfoTests.BasicProcessInfoTest() line 31
```

The target runtime instance has exited with exit code of 0xC0000005.

I have dumps for the target runtime instance if you are interested. I'll try to take a look and see if I can gleam anything.

Reply:
Exception:
```
Unhandled exception thrown: read access violation.
AppDomain::m_pTheAppDomain->m_pRootAssembly was nullptr.
```

Stack:
```
>	coreclr.dll!process_protocol_helper_get_process_info_2(_DiagnosticsIpcMessage * message, _DiagnosticsIpcStream * stream) Line 559	C++	Symbols loaded.
 	[Inline Frame] coreclr.dll!ds_process_protocol_helper_handle_ipc_message(_DiagnosticsIpcMessage *) Line 786	C++	Symbols loaded.
 	coreclr.dll!server_thread(void * data) Line 164	C++	Symbols loaded.
 	kernel32.dll!BaseThreadInitThunk()	Unknown	Non-user code. Symbols loaded.
 	ntdll.dll!RtlUserThreadStart()	Unknown	Non-user code. Symbols loaded.
```

Reply:
https://github.com/dotnet/runtime/issues/54083